backend/oasst_backend/error_codes.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+"""
+Open-Assistant backend API error codes.
```

Comment:
suggestion: make these into an IntEnum, so they're still int, but typing could be used

Reply:
yes I don't think any of that would change with an `IntEnum`. it would just be nicer to instead of having to take an `int` parameter, be able to take and `OAsstErrorCode` parameter (that behaves like an int)

Reply:
(regarding typing .. I normally put such an exception class in a shared lib, it is int to prevent dependencies between the base exception class and each and every error_code that would be defined). It would make sense to derive exception classes from it .. but I am not sure if the exception-filter type check in fast-api is polymorphic. I will try that.

Reply:
like this: https://docs.python.org/3/library/http.html#http.HTTPStatus

Reply:
I have now used a simple `IntEnum` for the error codes.

Important: There was an error in `api_auth()`, probably introduced when `DEBUG_SKIP_API_KEY_CHECK` was added,
see https://github.com/LAION-AI/Open-Assistant/blob/f208f65fafcac24f6c513dc80cbbcb4fb75178d4/backend/oasst_backend/api/deps.py#L55-L57 ... if the DB query returned None the function returned None (instead of raising an Unauthorized error...)

Reply:
To clarify: the `api_auth()` problem occurred when the `api_key` value sent with the request was not empty and both `settings.DEBUG_SKIP_API_KEY_CHECK == settings.DEBUG_SKIP_API_KEY_CHECK == False` and there was no matching or enabled api_client registered in the DB.